integration/insert_compat_test.go

```diff
@@ -111,17 +124,48 @@ func TestInsertCompat(t *testing.T) {
 	t.Parallel()
 
 	testCases := map[string]insertCompatTestCase{
-		"InsertEmptyDocument": {
-			insert: bson.D{},
-		},
 		"InsertIDArray": {
-			insert:     bson.D{{"_id", bson.A{"foo", "bar"}}},
+			insert:     []any{bson.D{{"_id", bson.A{"foo", "bar"}}}},
 			resultType: emptyResult,
 		},
 		"InsertIDRegex": {
-			insert:     bson.D{{"_id", primitive.Regex{Pattern: "^regex$", Options: "i"}}},
+			insert:     []any{bson.D{{"_id", primitive.Regex{Pattern: "^regex$", Options: "i"}}}},
 			resultType: emptyResult,
 		},
+
+		"InsertOrderedAllErrors": {
+			insert: []any{
+				bson.D{{"_id", bson.A{"foo", "bar"}}},
+				bson.D{{"_id", primitive.Regex{Pattern: "^regex$", Options: "i"}}},
+			},
+			ordered:    true,
+			resultType: emptyResult,
+		},
+		"InsertUnorderedAllErrors": {
+			insert: []any{
+				bson.D{{"_id", bson.A{"foo", "bar"}}},
+				bson.D{{"_id", primitive.Regex{Pattern: "^regex$", Options: "i"}}},
+			},
+			ordered:    false,
+			resultType: emptyResult,
+		},
+
+		"InsertOrderedOneError": {
+			insert: []any{
+				bson.D{{"_id", "1"}},
+				bson.D{{"_id", primitive.Regex{Pattern: "^regex$", Options: "i"}}},
+				bson.D{{"_id", "2"}},
```

Comment:
Okay, so here, if I understand correctly, both databases will:
1) insert _id: 1
2) return the error and don't rollback the first insert

and by the end our collection will look like: `[ {_id:1} ]` right?

Reply:
See https://github.com/FerretDB/FerretDB/issues/200#issuecomment-1003748090 and the whole issue in general

Reply:
@noisersup yes, correct. This is how MongoDB works (I checked manually, and the compat tests prove it).